core/kazoo_stdlib/test/kz_json_test.erl

```diff
@@ -832,13 +832,32 @@ set_value_test_() ->
                                                                                             ,{opt2,<<"my-opt2">>}
                                                                                             ])}
                                                          ])}])).
+-define(FROM_MAP_JSON_4_WRONG,
+        kz_json:from_list([{outer_key, kz_json:from_list([{inner_key,<<"inner_value">>}
+                                                         ,{inner_options, [{opt1,<<"my-opt1">>}
+                                                                          ,<<"anykey">>
+                                                                          ,{opt1,<<"my-opt1">>}
+                                                                          ]}
+                                                         ])}])).
+-define(FROM_MAP_JSON_4_RIGHT,
+        kz_json:from_list([{outer_key, kz_json:from_list([{inner_key,<<"inner_value">>}
+                                                         ,{inner_options, [kz_json:from_list([{opt1,<<"my-opt1">>}])
+                                                                          ,<<"anykey">>
+                                                                          ,kz_json:from_list([{opt1,<<"my-opt1">>}])
+                                                                          ]}
+                                                         ])}])).
 
 -define(FROM_MAP_MAP_1, #{options => [{opt1,<<"my-opt1">>},{opt2,<<"my-opt2">>}]}).
 -define(FROM_MAP_MAP_2, #{outer_key => #{inner_key => <<"inner_value">>}}).
 -define(FROM_MAP_MAP_3, #{outer_key => #{inner_key => <<"inner_value">>
                                         ,inner_options => [{opt1,<<"my-opt1">>}
                                                           ,{opt2,<<"my-opt2">>}
                                                           ]}}).
+-define(FROM_MAP_MAP_4, #{outer_key => #{inner_key => <<"inner_value">>
+                                        ,inner_options => [{opt1,<<"my-opt1">>}
```

Comment:
Is the intent here to be a proplist? In which case `<<"anykey">>` should be treated as if it was really `{<<"anykey">>, true}`?

Reply:
not a proplist, and that's exactly what we want to show
[{field, <<"account_id">>}
,<<"&realm=">>
,{field, <<"account_realm">>}
]

Reply:
I guess my thought is this list for inner_options doesn't have a valid JSON representation then.

Reply:
inner_options is a list of terms, when all elements are tuple, we transform it with from_list(), when they're not, we need to ensure that each element is a valid json element which a tuple is is not and must be transformed with from_list([tuple]) on the element.